### ISSUE TITLE
Characters can interact with each other modifier (BBCR)

### ISSUE THREAD

**Issue by AndrewScripter587** [Fasguy/BaldisBasicsModMenu-archive#97]
this could be a separate plugin because it's so complex.
you don't need to add this if it's too much.

**Why do you need this feature?**
Because it would be fun to see the characters fight amongst themselves

**Describe the feature**
TL;DR at the bottom

a toggle under each character's modifiers (and in bulk editor) that determines whether or not the character will interact with other characters based on them.
kind of like how beans can hit other characters with gum, but the character actively targets other characters.

when enabled, the characters will not only interact with each other, but also target them directly if they are not blind because of the blindness modifier.
i have a few ideas for how to make characters interact with each other:

For Playtime, my idea to make Playtime interact with other characters is once Playtime reaches them, they will just freeze for a certain amount of time. after they unfreeze, Playtime will not interact or target that character for the amount of time specified in the already existing cooldown modifier.

Beans: my idea for Beans is very simple, just whenever beans sees a character, Beans will do the same thing as when they see the player, but target the character instead.

Gotta Sweep will not have this modifier, as Gotta Sweep does not target anyone specifically and already interacts with all characters

Cloudy Copter will not have it either, for the same reason as Gotta Sweep.

Arts and Crafters, i don't really have an idea for. the other characters don't collect notebooks, so Arts and Crafters can't really teleport them. the only thing i can think of is that arts and crafters just runs away from the other characters like they do from the player when they have less than 7 notebooks.

1st Prize, just targets the other characters and pushes them like it does the player.
Principal of the Thing, just whenever they see a character breaking a rule(that they can break at all, like entering faculty rooms), does the same thing as when he sees a character when you've made them guilty of breaking a rule manually using the mod menu.

Chalkles, i don't really know how that will work. the only thing i can actually think of, is that when a character stays in a room that has a Chalkles chalk board in it for too long, Chalkles will do what he does when the player stays in there for too long, flying around the room and locking the door.

It's a Bully, i have a few ideas: the simpler one is to just force characters to go a different route to get past Its a Bully, and my other, less simple idea is when characters get close to Its a Bully, It's a Bully will first say one of the 2 lines, and then do something specific depending on what character is trying to go past: if Playtime tries to pass through, Playtime's jumprope will be cut and Playtime will go on cooldown, if Baldi tries to pass through, Baldi's ruler will break and the Broken Ruler event will immediately start, and any other character will be forced to move around It's a Bully, except for Principal of the Thing, who will just send It's a Bully to detention for bullying, regardless if the toggle for Principal of the Thing to interact with other characters is enabled, and Principal of the Thing will also scold It's a Bully when another character is near It's a Bully. 1st Prize will stop moving if they hit It's a Bully as if they were a wall, even making the "Bang!"(taken from in-game captions) sound and alerts Baldi if 1st Prize is moving fast enough when colliding. any characters being pushed or forcibly moved (like Mrs. Pomp, see below), through It's a Bully by any means such as Cloudy Copter, 1st Prize, Gotta Sweep, or Mrs. Pomp(see below), will stop being pushed/forcibly moved(unless they are one of the characters mentioned, then It's a Bully will do one of the mentioned things depending on what character they are).

Baldi and Null/Red Baldloon, grouped together because they do pretty much the same thing, would either just despawn or disable the instance of a character that they touch (unless Baldi is the one who collided with the other character and the Broken Ruler event is occurring, but Null/Red Baldloon would despawn or disable the instance regardless if the event is happening or not) and would also make one of the several jumpscare sounds (*BUZZ* from the captions) when they do.

Mrs. Pomp(normally unused but can be spawned with the mod menu), would run towards a character that they see, and would invite that character to Mrs. Pomp's class(not freezing time like Mrs. Pomp does when inviting the player), and then rush to their classroom as normal, and the invited character would instead of wandering randomly when they aren't heading to any specific spot(like Baldi or Null/Baldloon heading toward a sound), would instead head straight to Mrs. Pomp's classroom, and once they do, continue wandering. if the invited character doesn't make it in the amount of time Mrs. Pomp gives (changeable with the mod menu), Mrs. Pomp will wait for the invited character to enter a room if they aren't already in one, then run to the outside of the room, then wait for them to exit, then drags the invited character to Mrs. Pomp's room, lock the door, and run around in the room for a few seconds, then resumes normal activity.

Notes:
1. This (preferably) could be 2 separate toggles per character that act like the already existing Blind and Ignores Player Trigger modifiers, where one makes the character target others directly if on, but could still interact with them if this is off (they still target the player even if this is off), and the other allowing the character to interact with others if on (again, still can interact with the player even if off). if both are off, then they will not target nor interact with others, and if both are on, the character will target AND interact with others.
I say preferably because I would like to disable interaction with other characters for Baldi but still allow Baldi to target other characters and follow them around without having Baldi despawn them.

2. For characters that normally lock doors, such as Principal of the Thing, Chalkles, and Mrs. Pomp, they will still lock the door, but a separate modifier needs to be added that disallows characters that don't open locked doors (and instead just phase through them) from moving through locked doors unless the locked door is still opened for reasons like it was opened when it locked and hasn't closed yet, is open because of the flood event, or a character that can open locked doors(only Mrs. Pomp in BBCR) has opened it, when the modifier is enabled

TL;DR this would be a modifier that allows characters to interact with each other like they do with the player, and also target each other directly.

Edit: why were the replies below marked as spam? they were genuine replies that were not meant to be spam and are not spam in any way.

**Comment by ModernLucas** [comment]
WOW!!!! dude, i hope he add this, i Always wanted that! That's so AWESOME!!

**Comment by AndrewScripter587** [comment]
> WOW!!!! dude, i hope he add this, i Always wanted that! That's so AWESOME!!

Me too!

**Comment by AndrewScripter587** [comment]
why were the replies marked as spam?

**Comment by ModernLucas** [comment]
i don't know

**Comment by Fasguy** [comment]
This is better suited as a seperate mod. Afterwards a plugin could be created to enable integrating the ModMenu with this seperate mod.
